core/src/envelope.proto

```diff
@@ -0,0 +1,12 @@
+syntax = "proto2";
```

Comment:
Could you replace this file with https://github.com/libp2p/go-libp2p-core/blob/master/record/pb/envelope.proto ?